buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-							"[[slices]]\n"+
-								`paths = ["some-path", "some-other-path"]`+"\n",
-							filepath.Join(appDir, "launch-A-v1.toml"),
-						)
-
-						br, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						if err != nil {
-							t.Fatalf("Unexpected error:\n%s\n", err)
-						}
-
-						if s := cmp.Diff(br, buildpack.BuildResult{
-							LaunchBOM:   nil,
-							Labels:      []buildpack.Label{},
-							MetRequires: nil,
-							Processes:   []launch.Process{},
-							Slices:      []layers.Slice{{Paths: []string{"some-path", "some-other-path"}}},
-						}); s != "" {
-							t.Fatalf("Unexpected:\n%s\n", s)
-						}
-					})
-				})
-			})
-
-			when("the launch, cache and build flags are false", func() {
-				when("the flags are specified in <layer>.toml", func() {
-					it("renames <layers>/<layer> to <layers>/<layer>.ignore", func() {
-						h.Mkdir(t,
-							filepath.Join(layersDir, "A", "layer"),
-						)
-						h.Mkfile(t,
-							"[types]\n  build=false\n  cache=false\n  launch=false",
-							filepath.Join(layersDir, "A", "layer.toml"),
-						)
-
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNil(t, err)
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "A", "layer"))
-						h.AssertPathExists(t, filepath.Join(layersDir, "A", "layer.ignore"))
-					})
-				})
-
-				when("the flags aren't specified in <layer>.toml", func() {
-					it("renames <layers>/<layer> to <layers>/<layer>.ignore", func() {
-						h.Mkdir(t,
-							filepath.Join(layersDir, "A", "layer"),
-						)
-						h.Mkfile(t,
-							"",
-							filepath.Join(layersDir, "A", "layer.toml"),
-						)
-
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertNil(t, err)
-						h.AssertPathDoesNotExist(t, filepath.Join(layersDir, "A", "layer"))
-						h.AssertPathExists(t, filepath.Join(layersDir, "A", "layer.ignore"))
-					})
-				})
-			})
-		})
-
-		when("building succeeds with a clear env", func() {
-			it.Before(func() {
-				mockEnv.EXPECT().List().Return(append(os.Environ(), "TEST_ENV=cleared"))
-
-				bpTOML.Buildpack.Version = "v1.clear"
-				bpTOML.Dir = filepath.Join(buildpacksDir, "A", "v1.clear")
-				bpTOML.Buildpack.ClearEnv = true
-			})
-
-			it("should not apply user-provided env vars", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Error: %s\n", err)
-				}
-				if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-info-A-v1.clear")),
-					"TEST_ENV: cleared\n",
-				); s != "" {
-					t.Fatalf("Unexpected info:\n%s\n", s)
-				}
-			})
-
-			it("should set CNB_BUILDPACK_DIR", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Error: %s\n", err)
-				}
-				if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-buildpack-dir-A-v1.clear")),
-					bpTOML.Dir,
-				); s != "" {
-					t.Fatalf("Unexpected CNB_BUILDPACK_DIR:\n%s\n", s)
-				}
-			})
-		})
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L113-L169